### PR TITLE
[fix](Nereids) ComputePrecisionForRound should only process integral literal

### PR DESCRIPTION
**Pull request by morrySnow**

### CODE REVIEW

**Comment by morrySnow** [comment]
run buildall

**Comment by morrySnow** [comment]
run p0

**Review by starocean999 — APPROVED** [review]

File: fe/fe-core/src/main/java/org/apache/doris/nereids/NereidsPlanner.java [hunk 1]
```diff
@@ -581,8 +581,7 @@ public Optional<ResultSet> handleQueryInFe(StatementBase parsedStmt) {
             return Optional.of(resultSet);
         } else if (child instanceof PhysicalEmptyRelation) {
             List<Column> columns = Lists.newArrayList();
-            PhysicalEmptyRelation physicalEmptyRelation = (PhysicalEmptyRelation) physicalPlan.child(0);
-            for (int i = 0; i < physicalEmptyRelation.getProjects().size(); i++) {
+            for (int i = 0; i < physicalPlan.getOutput().size(); i++) {
```
Comment (zhiqiang-hhhh) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/NereidsPlanner.java:584]: what is the relation between this for loop with round function?

Comment (morrySnow) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/NereidsPlanner.java:584]: when we do computeprecision, we do not choose any signature, so its child do not cast to integer type, so child maybe any type

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/ComputePrecisionForRound.java [hunk 2]
```diff
@@ -37,7 +37,8 @@ default FunctionSignature computePrecision(FunctionSignature signature) {
             Expression floatLength = getArgument(1);
             int scale;
 
-            if (floatLength.isLiteral() || (floatLength instanceof Cast && floatLength.child(0).isLiteral()
+            if ((floatLength.isLiteral() && floatLength.getDataType() instanceof Int32OrLessType)
```
[the remark this answers is not in the document]
Comment (morrySnow) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/ComputePrecisionForRound.java:40]: when we do computePrecision, we do not finish compute signature. so type coercion do not apply. so its child could be any type literal

File: fe/fe-core/src/main/java/org/apache/doris/nereids/NereidsPlanner.java (hunk 1 above)
Comment (morrySnow) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/NereidsPlanner.java:584]: this is another bug, and have no relation with round function

**Comment by morrySnow** [comment]
already fixed by https://github.com/apache/doris/pull/34471

**Pull request closed without merging**